### DISCUSSION TITLE
Selectors:  I have questions regarding the do/do not portion of this document:  https://webdriver.io/docs/selectors

### DISCUSSION THREAD

**Discussion by kent-hervey** [webdriverio/webdriverio#11127] [Technical Discussion]
Can I ask my questions here?

Mostly they are around the rationale of why and why not and the meaning of some terms.

Ref:

> We do and do not recommend the following selectors:
> 
> Selector	Recommended	Notes
> $('button')	🚨 Never	Worst - too generic, no context.
> $('.btn.btn-large')	🚨 Never	Bad. Coupled to styling. Highly subject to change.
> $('#main')	⚠️ Sparingly	Better. But still coupled to styling or JS event listeners.
> $(() => document.queryElement('button'))	⚠️ Sparingly	Effective querying, complex to write.
> $('button[name="submission"]')	⚠️ Sparingly	Coupled to the name attribute which has HTML semantics.
> $('button[data-testid="submit"]')	✅ Good	Requires additional attribute, not connected to a11y.
> $('aria/Submit') or $('button=Submit')	✅ Always	Best. Resembles how the user interacts with the page.

**Comment by christian-bromann** [comment]
What is your particular question?

**Comment by kent-hervey** [comment]
Thank you for responding.

My first question is about ID.  In your example:  $('#main') ⚠️ Sparingly Better. But still coupled to styling or JS event listeners.

I have considered ID to be most preferred (always or almost always) when it exists for reasons including unique and unrelated to anything else.  Can you explain more why it is rated as sparingly? Including maybe how it is coupled to stylie and JS event listeners?

Thanks

**Reply by christian-bromann** [reply]
Well ids used to be known to fetch elements with JavaScript or as CSS selector. This might look totally different today. In fact I am not sure the usefulness of them other than using them as a reference for an internal document link. That said we believe that aria labels or test ids are a better choice and keep in mind that this list is just our recommendation and by no means you should oblige to it. If you find ids working for you, that's great, keep using them.

Some interesting articles about this:
- https://autify.com/blog/why-id-should-not-be-used/
- https://www.deque.com/blog/writing-automated-tests-accessibility/
